### DISCUSSION TITLE
Need help with parameter study on 10,000 different Voronoi meshes.

### DISCUSSION THREAD

**Discussion by gutza1** [idaholab/moose#24393] [Q&A Modules: General]
My team is attempting to run a parameter study that will generate 10,000 different meshes and perform a single time step on each. We have looked at the Parameter study tutorial on the website and have modified the code for our application, but when we attempted to run it, we got an error message claiming that the input file has no actionable object. What could be causing this error?

Second, how would one generate a discrete distribution of parameters (e.g. 1, 2, 3...)? We are attempting to do a parameter study where the parameter varied is the random seed number.

I have included the following code of the input file:

```
[ParameterStudy]
  input = MOOSE_10_grains_Stochastic.i
  parameters = 'UserObjects/voronoi/rand_seed'
  # quantities_of_interest = 'T_avg/value q_left/value'

  sampling_type = lhs
  num_samples = 10000
  distributions = 'uniform'
  uniform_lower_bound = 1
  uniform_upper_bound = 20000
[]
```

**Comment by GiudGiud** [comment]
@zachmprince

**Comment by zachmprince** [comment]
So the first issue is probably a bug in the way it automatically determines the `multiapp_mode` parameter. So I would just set this manually. The safest one is `multiapp_mode = batch-reset`, but I would need to see the input to suggest the best one. 

For the discrete sampling, there isn't a way to do this in the sampler. But one thing you can do is have this in MOOSE_10_grains_Stochastic.i:
```
rand_seed_real = 0.0 # Dummy value to be replaced by sampling
rand_seed = ${fparse int(rand_seed_real)}

[UserObjects]
  [voronoi]
    ...
    rand_seed = ${rand_seed} 
  []
[]
```
Then you would just perturb `rand_seed_real` in `ParameterStudy/parameters`:
```
[ParameterStudy]
  input = MOOSE_10_grains_Stochastic.i
  multiapp_mode = batch-reset
  parameters = 'rand_seed_real'
  # quantities_of_interest = 'T_avg/value q_left/value'

  sampling_type = lhs
  num_samples = 10000
  distributions = 'uniform'
  uniform_lower_bound = 1
  uniform_upper_bound = 20000
[]
```
Doing this requires `multiapp_mode = batch-reset`.

**Comment by gutza1** [comment]
Unfortunately, we have implemented your fixes and are getting the same error: "section "ParameterStudy" does not have an associated "Action.""

**Reply by GiudGiud** [reply]
How old is your moose version?
 Is your app including the stochastic tools module?

**Reply by gutza1** [reply]
We are using the latest version and have installed the app on the StochasticTools module. He have tried everything and we don't know what to do to fix the issue. We're at the end of our rope. If there anything you can suggest we do, please share it.

**Reply by GiudGiud** [reply]
What do you mean by installed the app on the module?

This does not seem right. To create an app, you need to use the stork script in moose/scripts and then enable every physics module you need in the Makefile

**Reply by gutza1** [reply]
Apologies, I misspoke. I meant that we had enabled the StochasticTools module while running the latest version.

**Reply by GiudGiud** [reply]
And your rebuilt (make) the executable after enabling the module right
what does `ldd <your_executable-opt>` return (on linux) ?

**Reply by gutza1** [reply]
`linux-vdso.so.1 =>  (0x00007fffedb02000)

libcoupling_xolotl_test-opt.so.0 => /home/muntmda/coupling_xolotl/test/lib/libcoupling_xolotl_test-opt.so.0 (0x00007fffed6d4000)

libcoupling_xolotl-opt.so.0 => /home/muntmda/coupling_xolotl/lib/libcoupling_xolotl-opt.so.0 (0x00007fffed48f000)

libmodule_loader_with_tm_pf-opt.so.0 => /home/muntmda/coupling_xolotl/moose/modules/module_loader/lib/libmodule_loader_with_tm_pf-opt.so.0 (0x00007fffed280000)

libphase_field-opt.so.0 => /home/muntmda/coupling_xolotl/moose/modules/phase_field/lib/libphase_field-opt.so.0 (0x00007fffec938000)

libtensor_mechanics-opt.so.0 => /home/muntmda/coupling_xolotl/moose/modules/tensor_mechanics/lib/libtensor_mechanics-opt.so.0 (0x00007fffeba6a000)

libmoose-opt.so.0 => /home/muntmda/coupling_xolotl/moose/framework/libmoose-opt.so.0 (0x00007fffe95a4000)

libpcre-opt.so.0 => /home/muntmda/coupling_xolotl/moose/framework/contrib/pcre/libpcre-opt.so.0 (0x00007fffe9380000)

libhit-opt.so.0 => /home/muntmda/coupling_xolotl/moose/framework/contrib/hit/libhit-opt.so.0 (0x00007fffe9122000)

libxolotlInterface.so => /home/muntmda/coupling_xolotl/xolotl/install/lib/libxolotlInterface.so (0x00007fffe8efa000)

libmesh_opt.so.0 => /home/muntmda/coupling_xolotl/moose/scripts/../libmesh/installed/lib/libmesh_opt.so.0 (0x00007fffe7921000)

libnetcdf.so.13 => /home/muntmda/coupling_xolotl/moose/scripts/../libmesh/installed/lib/libnetcdf.so.13 (0x00007fffe763a000)

libtimpi_opt.so.11 => /home/muntmda/coupling_xolotl/moose/scripts/../libmesh/installed/lib/libtimpi_opt.so.11 (0x00007fffe742d000)

libmpicxx.so.12 => /apps/local/mvapich2/2.3.7-1-gcc-8.4.0/lib/libmpicxx.so.12 (0x00007fffe720c000)

libslepc.so.3.16 => /home/muntmda/coupling_xolotl/moose/petsc/arch-moose/lib/libslepc.so.3.16 (0x00007fffe6ca1000)

libpetsc.so.3.16 => /home/muntmda/coupling_xolotl/moose/petsc/arch-moose/lib/libpetsc.so.3.16 (0x00007fffe5447000)

libHYPRE-2.23.0.so => /home/muntmda/coupling_xolotl/moose/petsc/arch-moose/lib/libHYPRE-2.23.0.so (0x00007fffe4e37000)

libstrumpack.so => /home/muntmda/coupling_xolotl/moose/petsc/arch-moose/lib/libstrumpack.so (0x00007fffe43f2000)

libsuperlu_dist.so.7 => /home/muntmda/coupling_xolotl/moose/petsc/arch-moose/lib/libsuperlu_dist.so.7 (0x00007fffe40f6000)

libhdf5_hl.so.200 => /home/muntmda/coupling_xolotl/moose/petsc/arch-moose/lib/libhdf5_hl.so.200 (0x00007fffe3ed3000)

libhdf5.so.200 => /home/muntmda/coupling_xolotl/moose/petsc/arch-moose/lib/libhdf5.so.200 (0x00007fffe388c000)

libparmetis.so => /home/muntmda/coupling_xolotl/moose/petsc/arch-moose/lib/libparmetis.so (0x00007fffe364e000)

libmetis.so => /home/muntmda/coupling_xolotl/moose/petsc/arch-moose/lib/libmetis.so (0x00007fffe33ed000)

libX11.so.6 => /lib64/libX11.so.6 (0x00007fffe30af000)

libmpifort.so.12 => /apps/local/mvapich2/2.3.7-1-gcc-8.4.0/lib/libmpifort.so.12 (0x00007fffe2e77000)

libmpi.so.12 => /apps/local/mvapich2/2.3.7-1-gcc-8.4.0/lib/libmpi.so.12 (0x00007fffe2031000)

libnuma.so.1 => /lib64/libnuma.so.1 (0x00007fffe1e25000)

libudev.so.1 => /lib64/libudev.so.1 (0x00007fffe1c0f000)

libxml2.so.2 => /lib64/libxml2.so.2 (0x00007fffe18a5000)

libibmad.so.12 => /lib64/libibmad.so.12 (0x00007fffe1685000)

librdmacm.so.1 => /lib64/librdmacm.so.1 (0x00007fffe1469000)

libibumad.so.3 => /lib64/libibumad.so.3 (0x00007fffe125f000)

libibverbs.so.1 => /lib64/libibverbs.so.1 (0x00007fffe1046000)

libpthread.so.0 => /lib64/libpthread.so.0 (0x00007fffe0e2a000)

libgfortran.so.5 => /apps/local/spack/software/gcc-4.8.5/gcc-8.4.0-jacdabugmghmzya2rpfqpym2bpzjomy2/lib64/libgfortran.so.5 (0x00007fffe09b9000)

libz.so.1 => /lib64/libz.so.1 (0x00007fffe07a3000)

libgcc_s.so.1 => /apps/local/spack/software/gcc-4.8.5/gcc-8.4.0-jacdabugmghmzya2rpfqpym2bpzjomy2/lib64/libgcc_s.so.1 (0x00007fffe058b000)

librt.so.1 => /lib64/librt.so.1 (0x00007fffe0383000)

libquadmath.so.0 => /apps/local/spack/software/gcc-4.8.5/gcc-8.4.0-jacdabugmghmzya2rpfqpym2bpzjomy2/lib64/libquadmath.so.0 (0x00007fffe0143000)

libstdc++.so.6 => /apps/local/spack/software/gcc-4.8.5/gcc-8.4.0-jacdabugmghmzya2rpfqpym2bpzjomy2/lib64/libstdc++.so.6 (0x00007fffdfdbf000)

libdl.so.2 => /lib64/libdl.so.2 (0x00007fffdfbbb000)

libm.so.6 => /lib64/libm.so.6 (0x00007fffdf8b9000)

libgomp.so.1 => /apps/local/spack/software/gcc-4.8.5/gcc-8.4.0-jacdabugmghmzya2rpfqpym2bpzjomy2/lib64/libgomp.so.1 (0x00007fffdf68b000)

libc.so.6 => /lib64/libc.so.6 (0x00007fffdf2bd000)

libxolotlSolver.so => /home/muntmda/coupling_xolotl/xolotl/install/lib/libxolotlSolver.so (0x00007fffdf007000)

libxolotlIO.so => /home/muntmda/coupling_xolotl/xolotl/install/lib/libxolotlIO.so (0x00007fffdede0000)

libxolotlViz.so => /home/muntmda/coupling_xolotl/xolotl/install/lib/libxolotlViz.so (0x00007fffdebcf000)

libxolotlCore.so => /home/muntmda/coupling_xolotl/xolotl/install/lib/libxolotlCore.so (0x00007fffde579000)

libxolotlPerf.so => /home/muntmda/coupling_xolotl/xolotl/install/lib/libxolotlPerf.so (0x00007fffde35b000)

libxolotlFactory.so => /home/muntmda/coupling_xolotl/xolotl/install/lib/libxolotlFactory.so (0x00007fffde14a000)

libxolotlOptions.so => /home/muntmda/coupling_xolotl/xolotl/install/lib/libxolotlOptions.so (0x00007fffddf1c000)

libboost_program_options.so.1.74.0 => /home/muntmda/coupling_xolotl/moose/petsc/arch-moose/lib/libboost_program_options.so.1.74.0 (0x00007fffddcb4000)

libxolotlUtil.so => /home/muntmda/coupling_xolotl/xolotl/install/lib/libxolotlUtil.so (0x00007fffdda78000)

libboost_log.so.1.74.0 => /home/muntmda/coupling_xolotl/moose/petsc/arch-moose/lib/libboost_log.so.1.74.0 (0x00007fffdd7ab000)

libboost_filesystem.so.1.74.0 => /home/muntmda/coupling_xolotl/moose/petsc/arch-moose/lib/libboost_filesystem.so.1.74.0 (0x00007fffdd58d000)

libboost_log_setup.so.1.74.0 => /home/muntmda/coupling_xolotl/moose/petsc/arch-moose/lib/libboost_log_setup.so.1.74.0 (0x00007fffdd2ef000)

libboost_thread.so.1.74.0 => /home/muntmda/coupling_xolotl/moose/petsc/arch-moose/lib/libboost_thread.so.1.74.0 (0x00007fffdd0d0000)

libboost_date_time.so.1.74.0 => /home/muntmda/coupling_xolotl/moose/petsc/arch-moose/lib/libboost_date_time.so.1.74.0 (0x00007fffdcece000)

libboost_regex.so.1.74.0 => /home/muntmda/coupling_xolotl/moose/petsc/arch-moose/lib/libboost_regex.so.1.74.0 (0x00007fffdcc19000)

libboost_chrono.so.1.74.0 => /home/muntmda/coupling_xolotl/moose/petsc/arch-moose/lib/libboost_chrono.so.1.74.0 (0x00007fffdca10000)

libboost_atomic.so.1.74.0 => /home/muntmda/coupling_xolotl/moose/petsc/arch-moose/lib/libboost_atomic.so.1.74.0 (0x00007fffdc809000)

libkokkoscontainers.so.3.7 => /home/muntmda/coupling_xolotl/kokkos/install/lib64/libkokkoscontainers.so.3.7 (0x00007fffdc606000)

libkokkoscore.so.3.7 => /home/muntmda/coupling_xolotl/kokkos/install/lib64/libkokkoscore.so.3.7 (0x00007fffdc36d000)

libkokkossimd.so.3.7 => /home/muntmda/coupling_xolotl/kokkos/install/lib64/libkokkossimd.so.3.7 (0x00007fffdc16b000)

/lib64/ld-linux-x86-64.so.2 (0x00007fffed8e3000)

libxcb.so.1 => /lib64/libxcb.so.1 (0x00007fffdbf43000)

libcap.so.2 => /lib64/libcap.so.2 (0x00007fffdbd3e000)

libdw.so.1 => /lib64/libdw.so.1 (0x00007fffdbaed000)

liblzma.so.5 => /lib64/liblzma.so.5 (0x00007fffdb8c7000)

libnl-route-3.so.200 => /lib64/libnl-route-3.so.200 (0x00007fffdb65a000)

libnl-3.so.200 => /lib64/libnl-3.so.200 (0x00007fffdb439000)

libXau.so.6 => /lib64/libXau.so.6 (0x00007fffdb235000)

libattr.so.1 => /lib64/libattr.so.1 (0x00007fffdb030000)

libelf.so.1 => /lib64/libelf.so.1 (0x00007fffdae18000)

libbz2.so.1 => /lib64/libbz2.so.1 (0x00007fffdac08000)`

**Reply by GiudGiud** [reply]
This doesn’t include stochastic tools

can you please paste your make file here

**Reply by gutza1** [reply]
Here is the makefile, sorry for the delay.

[Makefile.txt](https://github.com/idaholab/moose/files/11556077/Makefile.txt)

**Reply by GiudGiud** [reply]
can you `git clean -xfd` then rebuild?

you have the right command in the Makefile to enable stochastic_tools

**Reply by gutza1** [reply]
Apologies for the delay. We've been trying to figure out other solutions to the problem, but after seeing a past user who encountered a similar issue, I'm wondering if the problem lies in the base script itself. I've posted it here, please take a look to see if there's anything that wouldn't play well with StochasticTools.

[MOOSE_10_grains_Stochastic.txt](https://github.com/idaholab/moose/files/11619049/MOOSE_10_grains_Stochastic.txt)

**Reply by GiudGiud** [reply]
@zachmprince 

is the registering issue fixed now though?

**Reply by zachmprince** [reply]
The only thing I notice in your input is that you should remove this:
```
[StochasticTools]
[]
```

This block is only relevant in your driving STM input and only if you are not using `ParameterStudy`. Also, if you are using `ParameterStudy`, you can remove this block as well:
```
[Controls/stochastic]
  type = SampleReceiver
[]
```

Other than that, everything else seems fine.

**Comment by gutza1** [comment]
@zachmprince
